modules/tinymce/src/themes/silver/main/ts/ui/skin/Loader.ts

```diff
@@ -14,40 +15,72 @@ const loadStylesheet = (editor: Editor, stylesheetUrl: string, styleSheetLoader:
-  return loadStylesheet(editor, skinUiCss, editor.ui.styleSheetLoader);
+const loadRawCss = (editor: Editor, key: string, css: string, styleSheetLoader: StyleSheetLoader): void => {
+  // Ensure the stylesheet is cleaned up when the editor is destroyed
+  editor.on('remove', () => styleSheetLoader.unloadRawCss(key));
+  return styleSheetLoader.loadRawCss(key, css);
+};
+
+const loadUrlUiSkins = (editor: Editor, skinUrl: string): Promise<void> => {
+  const skinUiCss = skinUrl + '/skin.css';
+  const css = EditorManager.resources.get(skinUiCss);
+
+  if (Type.isString(css)) {
+    loadRawCss(editor, skinUiCss, css, editor.ui.styleSheetLoader);
+    return Promise.resolve();
+  } else {
+    const skinUiCss = skinUrl + '/skin.min.css';
+    return loadStylesheet(editor, skinUiCss, editor.ui.styleSheetLoader);
+  }
 };
 
-const loadShadowDomUiSkins = (editor: Editor, skinUrl: string): Promise<void> => {
+const loadShadowDomUrlUiSkins = (editor: Editor, skinUrl: string): Promise<void> => {
   const isInShadowRoot = SugarShadowDom.isInShadowRoot(SugarElement.fromDom(editor.getElement()));
   if (isInShadowRoot) {
-    const shadowDomSkinCss = skinUrl + '/skin.shadowdom.min.css';
-    return loadStylesheet(editor, shadowDomSkinCss, DOMUtils.DOM.styleSheetLoader);
+
+    const shadowDomSkinCss = skinUrl + '/skin.shadowdom.css';
+    const css = EditorManager.resources.get(shadowDomSkinCss);
+
+    if (Type.isString(css)) {
+      loadRawCss(editor, shadowDomSkinCss, css, DOMUtils.DOM.styleSheetLoader);
+      return Promise.resolve();
+    } else {
+      const shadowDomSkinCss = skinUrl + '/skin.shadowdom.min.css';
+      return loadStylesheet(editor, shadowDomSkinCss, DOMUtils.DOM.styleSheetLoader);
+    }
   } else {
     return Promise.resolve();
   }
 };
 
-const loadSkin = (isInline: boolean, editor: Editor): Promise<void> => {
+const loadUrlSkin = (isInline: boolean, editor: Editor): Promise<void> => {
   const skinUrl = Options.getSkinUrl(editor);
-
   if (skinUrl) {
-    editor.contentCSS.push(skinUrl + (isInline ? '/content.inline' : '/content') + '.min.css');
+    const css = EditorManager.resources.get(skinUrl + (isInline ? '/content.inline' : '/content') + '.css');
+
+    if (Type.isString(css)) {
+      loadRawCss(editor, skinUrl, css, editor.ui.styleSheetLoader);
+    } else {
+      editor.contentCSS.push(skinUrl + (isInline ? '/content.inline' : '/content') + '.min.css');
+    }
   }
 
   // In Modern Inline, this is explicitly called in editor.on('focus', ...) as well as in render().
   // Seems to work without, but adding a note in case things break later
   if (!Options.isSkinDisabled(editor) && Type.isString(skinUrl)) {
     return Promise.all([
-      loadUiSkins(editor, skinUrl),
-      loadShadowDomUiSkins(editor, skinUrl)
-    ]).then(SkinLoaded.fireSkinLoaded(editor), SkinLoaded.fireSkinLoadError(editor, 'Skin could not be loaded'));
+      loadUrlUiSkins(editor, skinUrl),
+      loadShadowDomUrlUiSkins(editor, skinUrl)
```

Comment:
Do they need to be renamed now? Perhaps keep the old names so the diff is easier to read 🤔

Reply:
Probably not. I'll change them back.